test/validation/api/traffic_mngr/traffic_mngr.c

```diff
@@ -4309,6 +4309,116 @@ static void traffic_mngr_test_queue_stats(void)
 	CU_ASSERT(odp_tm_is_idle(odp_tm_systems[0]));
 }
 
+static int traffic_mngr_check_node_stats(void)
+{
+	int i;
+
+	for (i = 0; i < NUM_LEVELS; i++) {
+		/* Report test case as active if even one level supports
+		 * stats.
+		 */
+		if (tm_capabilities.per_level[i].node_stats.all_counters)
+			return ODP_TEST_ACTIVE;
+	}
+
+	return ODP_TEST_INACTIVE;
+}
+
+static void traffic_mngr_test_node_stats(void)
+{
+	odp_tm_node_stats_t stats_start[NUM_LEVELS];
+	odp_tm_node_stats_capability_t stats_capa;
+	odp_tm_node_t nodes[NUM_LEVELS];
+	odp_tm_node_stats_t stats_stop;
+	odp_tm_queue_info_t q_info;
+	odp_tm_node_info_t n_info;
+	odp_tm_queue_t tm_queue;
+	odp_tm_capabilities_t capa;
+	pkt_info_t pkt_info;
+	uint32_t pkts_sent;
+	uint32_t num_pkts = MIN(50, MAX_PKTS);
+	uint32_t pkt_len = 256;
+	int i;
+
+	memset(stats_start, 0, sizeof(stats_start));
+	memset(&stats_stop, 0, sizeof(stats_stop));
```

Comment:
Maybe it would be better to remove these. They may hide a scenario where the implementation doesn't set unsupported fields to zero.